store/smartAccount/createSmartAccount.ts

```diff
@@ -0,0 +1,163 @@
+import { Alert } from 'react-native';
+import * as Passkeys from 'react-native-passkeys';
+import { utf8StringToBuffer, bufferToBase64URLString } from '@/utils/base64';
+import { ToCoinbaseSmartAccountReturnType, WebAuthnAccount, createBundlerClient, toCoinbaseSmartAccount } from 'viem/account-abstraction';
+import { Address, createPublicClient, hashMessage, hashTypedData, Hex, http, SignableMessage } from 'viem';
+import { base } from 'viem/chains';
+import type { SignReturnType } from 'webauthn-p256';
+
+export function setupViem() {
+  const rpcUrl = 'https://mainnet.base.org';
+  const bundlerUrl = 'https://public.pimlico.io/v2/1/rpc';
+
+  const publicClient = createPublicClient({
+    chain: base,
+    transport: http(rpcUrl),
+  });
+
+  const bundlerClient = createBundlerClient({
+    client: publicClient,
+    transport: http(bundlerUrl),
+    paymaster: true,
+  });
+
+  return { publicClient, bundlerClient };
+}
+
+export async function createSmartAccount(username: string): Promise<{
+  smartAccount: ToCoinbaseSmartAccountReturnType;
+  address: Address;
+}> {
+  try {
+    // TODO: ideally this should not be done here
+    const { publicClient, bundlerClient } = setupViem();
+
+    const isPasskeyAvailable = Passkeys.isSupported();
+    if (!isPasskeyAvailable) {
+      throw new Error('Passkeys are not available on this device');
+    }
+
+    // const options = await this.api.getRegistrationOptions(username);
+
+    // const registrationResponse = await this.passKeyImpl.createPassKeyCredential(options);
+
+    const credential = await Passkeys.create({
+      challenge: bufferToBase64URLString(utf8StringToBuffer('mock-challenge')),
+      rp: {
+        name: 'Liquid Smart Account',
+        id: 'api.useliquid.xyz',
+      },
+      user: {
+        id: bufferToBase64URLString(utf8StringToBuffer(username)),
+        name: username,
+        displayName: username,
+      },
+      pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      authenticatorSelection: {
+        authenticatorAttachment: 'platform',
+        userVerification: 'required',
+        residentKey: 'required',
+        requireResidentKey: true,
+      },
+    });
```

Comment:
this will be updated in the pr for API calls right? @renanmav

Reply:
Yes @njokuScript

Reply:
LGTM!